src/pages/keystone-test.tsx

```diff
@@ -0,0 +1,394 @@
+  const router = useRouter()
+  const { user } = useUser()
+  const { loading, error, data } = useMySpaceQuery()
+  const { trackEvent } = useAnalytics()
+
+  const [sortBy, setSort] = useState<SortBy>('SORT_TYPE')
+  const [selectMode, setSelectMode] = useState<boolean>(false)
+  const [selectedCollections, setSelectedCollections] =
+    useState<SelectedCollections>([])
+  const [flash, setFlash] = useState<React.ReactNode>(null)
+
+  const [handleAddCollections] = useAddCollectionsMutation()
+  const [handleAddCollection] = useAddCollectionMutation()
+  const [handleAddBookmark] = useAddBookmarkMutation()
+  const {
+    loading: cmsCollectionsLoading,
+    error: cmsCollectionsError,
+    data: cmsCollections,
+  } = useQuery(GET_KEYSTONE_COLLECTIONS, {
+    context: {
+      clientName: 'cms',
+    },
+  })
+
+  const collections = cmsCollections?.collections as CollectionRecords
+
+  const {
+    loading: cmsBookmarksLoading,
+    error: cmsBookmarksError,
+    data: cmsBookmarks,
+  } = useQuery(GET_KEYSTONE_BOOKMARKS, {
+    context: {
+      clientName: 'cms',
+    },
+  })
+
+  const bookmarks = cmsCollections?.bookmarks as BookmarkRecords
+
+  useEffect(() => {
+    if (router.query.selectMode == 'true') {
+      setSelectMode(true)
+    }
+  }, [router.query])
+
+  if (error) return <p>Error</p>
+
+  const userCollections: CollectionType[] =
+    (data &&
+      data.mySpace.filter((w): w is CollectionType => isCollection(w))) ||
+    []
+  const collectionsLength = userCollections.length || 0
+
+  const remainingSections =
+    MAXIMUM_COLLECTIONS - (collectionsLength + selectedCollections.length)
+
+  const canAddSections = collectionsLength < MAXIMUM_COLLECTIONS
+
+  const handleSortClick = (sortType: SortBy) => {
+    const sortTypeAction =
+      sortType === 'SORT_TYPE' ? 'Sort by type' : 'Sort alphabetically'
+    trackEvent('S&A sort', sortTypeAction)
+    setSort(sortType)
+  }
+
+  const handleToggleSelectMode = () => {
+    setSelectMode((currentMode) => !currentMode)
+    setSelectedCollections([])
+  }
+
+  const widgetClasses = classnames(styles.widgetContainer, {
+    [styles.selectMode]: selectMode,
+  })
+
+  const handleSelectCollection = (id: string): void => {
+    if (isSelected(id)) {
+      setSelectedCollections((state) => {
+        const itemIndex = state.indexOf(id)
+        const newState = [...state]
+        newState.splice(itemIndex, 1)
+        return newState
+      })
+    } else {
+      setSelectedCollections((state) => [...state, id])
+    }
+  }
+
+  const isSelected = (id: string): boolean =>
+    selectedCollections.indexOf(id) > -1
+
+  const handleAddSelected = () => {
+    const collectionObjs = selectedCollections.map((id) =>
+      collections.find((i) => i.id === id)
+    ) as CollectionRecords
+
+    const collectionTitles = collectionObjs.map((c) => c.title).join(',')
+    trackEvent('S&A add collection', 'Add selected', collectionTitles)
+
+    handleAddCollections({
```

Comment:
Trying to add collections from this page gives me a GraphQL error, it looks like the Keystone API is returning `__typename` which this mutation is not expecting as part of the object.

Reply:
I was actually just looking into this on my branch. Should the change be made on this branch, or could I just update the CollectionRecord type to include the `__typename` field?